ui/src/utils/visualisations/aggregationGroup/grouper/index.js

```diff
@@ -78,38 +78,69 @@ const getExistsMatch = ({ valueType, groupType, valueOpCase }) => {
   }
 };
 
-const getUnwind = ({ groupType }) => {
-  if (groupType.startsWith('statement.context.contextActivities.')) {
-    return { path: `$${groupType}` };
-  }
-  return null;
+const createPersonaAttributeStages = ({ groupType }) => {
+  const lookupStage = createStagePipeline('$lookup', {
+    from: 'personaAttributes',
+    as: 'personaAttrs',
+    localField: 'person._id',
+    foreignField: 'personaId'
+  });
+
+  const unwindStage = createStagePipeline('$unwind', {
+    path: '$personaAttrs'
+  });
+
+  const matchAttrsKeyStage = createStagePipeline('$match', {
+    'personaAttrs.key': groupType.split('.')[2] || ''
+  });
+
+  return lookupStage
+    .concat(unwindStage)
```

Comment:
I think an additional match stage before the unwind will actually filter any people's records that don't have the attribute prior to unwinding. Debatable if this make the following two steps quicker or not

```suggestion
    .concat(matchAttrsKeyStage)
    .concat(unwindStage)
```

Reply:
OK. I'll measure performance with the fake data.